packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding_phone_confirmation/blocs/onboarding_phone_bloc.dart

```diff
@@ -96,14 +112,37 @@ class OnboardingPhoneBloc extends $OnboardingPhoneBloc {
       .map((args) => _numberValidatorService.validateNumberAndCountryCode(
           args.$1, args.$2))
       .startWith('')
-      .asBroadcastStream();
+      .shareReplay(maxSize: 1);
+
+  String? _validateAndFormatPhoneNumber(
+    Result<CountryCodeModel?> countryCodeResult,
+    Result<String> phoneResult,
+  ) {
+    if (countryCodeResult is ResultError || phoneResult is ResultError) {
+      return null;
+    }
+    if (countryCodeResult is ResultLoading || phoneResult is ResultLoading) {
+      return null;
+    }
+
+    final code =
+        (countryCodeResult as ResultSuccess<CountryCodeModel?>).data?.code ??
+            '';
+    final phone = (phoneResult as ResultSuccess<String>).data;
+
+    return '+$code $phone';
```

Comment:
To reduce the code and ensure stability let's first check whether the result are of type success, otherwise we can return a null

Reply:
Done.